cmd/dockerd/daemon.go

```diff
@@ -484,7 +484,7 @@ func initRouter(s *apiserver.Server, d *daemon.Daemon, c *cluster.Cluster) {
 		image.NewRouter(d, decoder),
 		systemrouter.NewRouter(d, c),
 		volume.NewRouter(d),
-		build.NewRouter(dockerfile.NewBuildManager(d)),
+		build.NewRouter(manager.NewBuildManager(d, d), d),
```

Comment:
Currently, the glue code between components does exist in the api. You can see many examples of this, for example https://github.com/moby/moby/blob/0e9148b1e92dc6cf79dc57c71423297d5c11d0ca/api/server/router/network/network_routes.go#L51-L58

https://github.com/moby/moby/blob/0e9148b1e92dc6cf79dc57c71423297d5c11d0ca/api/server/router/system/system_routes.go#L38-L40

and I think it is currently the best place to put them. When builder starts caching directories, it will opt into `df/prune` API in the same way. The HTTP API does not follow the component boundaries. We could separate out the actual HTTP parsing to a separate pkg but I'm not sure if this would be worth it as these cases are quite simple.

> "squash" definitely doesn't belong in the "core builder"

Can you explain why you think that. If I define a build as a function from build definition to a result then squash is part of that as the result of that build is the squashed image.